server/server_linux.go

```diff
@@ -139,3 +141,47 @@ func configureMaxThreads() error {
 	logrus.Debugf("Golang's threads limit set to %d", maxThreads)
 	return nil
 }
+
+// checkKernelRROMountSupport checks the kernel support for the Recursive Read-only (RRO) mounts.
+func checkKernelRROMountSupport() error {
+	// The minimal required kernel version is 5.12 or newer.
+	const kernelVersion = 512
+
+	checkKernelRROSupport := func() error {
```

Comment:
BTW I think I'd prefer this inline below, rather than saving the functor and calling it below. I don't think we save on indentation or code readability as-is

Reply:
Sure! Will do.

Reply:
Done.